x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlResponseListener.java

```diff
@@ -154,23 +154,29 @@ private RestResponse buildResponse(EsqlQueryResponse esqlResponse) throws IOExce
     }
 
     /**
-     * Log the execution time and query when handling an ES|QL response.
+     * Log internal server errors all the time and log queries if debug is enabled.
      */
     public ActionListener<EsqlQueryResponse> wrapWithLogging() {
+        ActionListener<EsqlQueryResponse> listener = ActionListener.wrap(this::onResponse, ex -> {
+            logOnFailure(LOGGER, ex);
+            onFailure(ex);
+        });
+        if (LOGGER.isDebugEnabled() == false) {
+            return listener;
+        }
         return ActionListener.wrap(r -> {
-            onResponse(r);
+            listener.onResponse(r);
             // At this point, the StopWatch should already have been stopped, so we log a consistent time.
-            LOGGER.info(
+            LOGGER.debug(
                 "Finished execution of ESQL query.\nQuery string: [{}]\nExecution time: [{}]ms",
                 esqlQuery,
                 stopWatch.stop().getMillis()
             );
         }, ex -> {
             // In case of failure, stop the time manually before sending out the response.
             long timeMillis = stopWatch.stop().getMillis();
-            LOGGER.info("Failed execution of ESQL query.\nQuery string: [{}]\nExecution time: [{}]ms", esqlQuery, timeMillis);
-            logOnFailure(LOGGER, ex);
-            onFailure(ex);
+            LOGGER.debug("Failed execution of ESQL query.\nQuery string: [{}]\nExecution time: [{}]ms", esqlQuery, timeMillis);
```

Comment:
should we log atleast at info for exceptions probably excluding the  esqlQuery,? Also i feel the new block in beginning is redundant to this, can we just keep a condition. only at LOGGER.debug level?

Reply:
> should we log atleast at info for exceptions probably excluding the esqlQuery,?

We do that already actually. That's what `logOnFailure` does, though it's based on the type of exception. An exception that generates a 500-level response will get logged without the query at `WARN` level. This was always a duplicate of that which is a little silly, but not really hurting anything at the moment.

> Also i feel the new block in beginning is redundant to this, can we just keep a condition. only at LOGGER.debug level?

I've done it this way to make it really obvious to the reader that there's special debug logging things - and that they shouldn't run unless we have it enabled. We do this kind of thing if the debug logging itself is slow, but in this case it's mostly just slow to read. But doing it this way means you don't really have to carefully read the debugging side to know it's not doing expensive things.